### DISCUSSION TITLE
How do your Docs work?

### DISCUSSION THREAD

**Discussion by thewatts** [ddnexus/pagy#534] [Q&A] [answered]
Hi there!

1. Pagy is FANTASTIC! Thanks for all you've done to make is such a great gem!

2. Your docs (the website generated from your /docs folder) are also wonderful.

May I ask: What are you doing to generate that website from that directory?

Is it something that GitHub provides out of the box? Have you had to customize anything? Is there a blog post / resource that details the process?

I'd love to use the same methodology for some projects I'm a part of.

Many thanks for your help!

**Comment by benkoshy** [comment, marked as answer]
> May I ask: What are you doing to generate that website from that directory?

https://retype.com/

> Is it something that GitHub provides out of the box? Have you had to customize anything? Is there a blog post / resource that details the process?

There is no blog post on the pagy website. However it is very simple:

Pagy uses a github action to trigger the build process. Once built, then [Github pages](https://pages.github.com/) takes over from there - you just have to set it to point to the correct branch which only the admin can do: in our case it is the `docs-site` branch.

https://github.com/ddnexus/pagy/blob/master/.github/workflows/retype-action.yml

More detailed docs on the retype website: https://retype.com/guides/github-actions/

Pagy repo is using the Open Source version, subject to certain limitations (i.e. total page count). Pro version opens up other benefits: https://retype.com/pro/

I cannot be more effusive in my praise of geoffreymcgill (and perhaps others) who have made it available to Pagy.

**Reply by thewatts** [reply]
Thank you so much!

**Reply by geoffreymcgill** [reply]
Thanks for kind words about Retype!

Feel free to contact me if you have any questions or feedback regarding Retype and I will do my best to assist.
